### ISSUE TITLE
Unexported Identifier Not Caught at Compile Time

### ISSUE THREAD

**Issue by maxsnew** [elm/compiler#394]
Say you have 2 files:

A.elm:

``` Haskell
module A (bar) where
foo = asText 3
bar = asText 4
```

Main.elm:

``` Haskell
module Main where
import open A
main = foo
```

This should be a compile-time error "Could not find variable foo", but it compiles fine and instead you get a very unfriendly runtime error.

However if you have this Main.elm:

``` Haskell
module Main where
import open A
main = qux
```

You get the desired error: "Could not find variable 'qux'." So it seems the compiler is keeping unexported identifiers around when it checks for unbound variables.
